### DISCUSSION TITLE
Add way to show vertical lines as decorations

### DISCUSSION THREAD

**Discussion by rhult** [patrykandpatrick/vico#694] [Ideas]
I need a way to show a vertical line in a line chart, that is not connected to the series at all. Although I can add an extra series with just this value and a persistent marker, this affects the axis item placer for the x axis, resulting in different selection of x axis labels (I think because Vico considers all the series).

I can probably implement such a decoration myself, but I expect more people to find this useful, so it might make sense to have support built-in for it.

**Comment by patrickmichalik** [comment]
Hello! Thanks for the suggestion. This is a planned addition—please see #615. In the meantime, indeed, you can implement this as your own `Decoration`.

With some modifications, the first approach that you’ve described can work too. If the _x_ value to be highlighted isn’t a multiple of the current _x_ step, you can lock the _x_ step at the current value via `getXStep`. Similarly, if the _x_ value to be highlighted lies outside of the current _x_ range, you can lock the _x_ range at the current value via `AxisValueOverrider`. With no changes to the _x_ step and _x_ range, the `HorizontalAxis` will remain unaffected. That said, creating a `Decoration` would be optimal. Let me know if you have any questions.
